### ISSUE TITLE
ODROID XU4 - cannot set resolution

### ISSUE THREAD

**Issue by hooby3dfx** [libretro/Lakka#477]
I have a 720p native resolution TV which also accepts 1080i input.

The Lakka image always seems to boot to 1080i regardless of whether I follow the instructions to update the boot.ini file. This results in poor performance at 30 Hz. 
I would like to force it to 720p.
My TV is a little funky as described here http://libretro.com/forums/showthread.php?t=6803&p=46164 however the official ODROID Android image correctly boots to 720p.

Possible duplicate of https://github.com/libretro/Lakka/issues/261

Outside of the boot.ini file, is there any other way to set the resolution? The retroarch fullscreen settings in the cfg dont seem to work either.

**Comment by hooby3dfx** [comment]
Oh WTF!
The last line of the included boot.ini does not pass the video_output param!!
Hence following the instructions would have no effect XD
```
#------------------------------------------------------------------------------------------------------
#
# Kernel boot
#
#------------------------------------------------------------------------------------------------------
setenv bootargs "${console} ${bootrootfs} ${openelec} smsc95xx.macaddr=${macaddr}"
```
should be
```
#------------------------------------------------------------------------------------------------------
#
# Kernel boot
#
#------------------------------------------------------------------------------------------------------
setenv bootargs "${console} ${bootrootfs} ${video_output} ${openelec} smsc95xx.macaddr=${macaddr}"

```


SO, now that I can actually set the resolution I am getting this type of output:
![img_20160829_233311654](https://cloud.githubusercontent.com/assets/516889/18075250/22e36bec-6e41-11e6-9d79-b06d0893c5bd.jpg)

**Comment by cudencuden** [comment]
for retroarch.cfg changes, did you stop retroarch service before making changes? systemctl stop retroarch

**Comment by Kivutar** [comment]
Ah you're right! Well spotted. I need to change this. I will do just after I'm finished with the LibreELEC port.

Sadly I have no idea why you are getting this kind of signal.

Can you try the fbset command line in this script? https://github.com/zalaare/OpenELEC-Odroid/blob/odroid-6.0/projects/Odroid/devices/C1/initramfs/platform_init

**Comment by hooby3dfx** [comment]
I found something else interesting. I dont think the issue is with my TV.
After getting the 720p mode set, I plugged the odroid into my other
display, which also got the distorted 720p output.

I think there is an issue with the Lakka image causing 720p output to have
the problem all the time on the xu4 at least (don't have an xu3).

Can you reproduce it now by forcing 720p with the fixed boot.ini?

I will also try and fiddle with fbset; which command in particular?

**Comment by Kivutar** [comment]
Could be related http://forum.odroid.com/viewtopic.php?f=94&t=15840

**Comment by hooby3dfx** [comment]
But that doesn't explain why it would work on Android?

**Comment by hooby3dfx** [comment]
Can you point me in the direction of the code/script that handles initializing the settings for the graphics mode in Lakka (at least for odroid xu3/4)?

**Comment by hooby3dfx** [comment]
Ah ha! Here is another thread of someone else with what looks like the same issue from the photo:
http://forum.odroid.com/viewtopic.php?f=99&t=22716

As a temporary work around, it looks like 1024x768 works.

**Comment by Kivutar** [comment]
We do have the commit he's talking about in that thread.

So it's a triple buffering issue?

**Comment by hooby3dfx** [comment]
What do you think the next step is here?

If anyone else with an ODROID XU4 sees this, can you force 720p and see if you get the same output issue on your TV/monitor?

**Comment by Kivutar** [comment]
I think the next step is to experiment with fbset.

You have some examples here in the C1 project https://github.com/libretro/Lakka/blob/lakka/projects/OdroidC1/initramfs/platform_init

**Comment by hooby3dfx** [comment]
Is there a way confirm the kernel source (like the commit hash) that was used?
The latest stable Lakka build outputs this for the version:
```
Lakka:~ # uname -a
Linux Lakka 3.10.96 #1 SMP PREEMPT Wed Sep 14 06:27:12 CEST 2016 armv7l GNU/Linux
Lakka:~ # cat /proc/version
Linux version 3.10.96 (kivutar@gcc20) (gcc version 4.9.2 (GCC) ) #1 SMP PREEMPT Wed Sep 14 06:27:12 CEST 2016
```
Shouldn't that be higher?

**Comment by Kivutar** [comment]
The commit id https://github.com/libretro/Lakka/blob/lakka/packages/linux/package.mk#L34

**Comment by hooby3dfx** [comment]
Thanks - a couple updates:
1. I found that the issue does not seem to occur on OpenELEC-6.0.2.0 (Linux OpenELEC 3.10.82+30cd824 from http://forum.odroid.com/viewtopic.php?f=79&t=5915)
2. The issue appears during boot of the official hardkernel linux release (3.10.103; http://odroid.com/dokuwiki/doku.php?id=en:xu3_ubuntu_release_note_20161011)
3. Issue occurs in Lakka build as far back as Lakka-OdroidXU3.arm-devel-20160109151244-r21276-g4dadcab

**Comment by Enverex** [comment]
I'm getting the same when trying to use a protector on my Odroid-XU4 with Lakka (the projector's native res is 1280x800). Seems to happen at 720p too (forced via boot.ini).

Is this basically unusable except at 1080p?

**Comment by hooby3dfx** [comment]
That is what I believe.

**Comment by hooby3dfx** [comment]
@Enverex the issue seems to just be with 720p. a temp workaround is actually to force 1024x768 via the boot file.

Update on this issue - after a little more testing, I confirmed that the problem exists upstream and even on the current official odroid Ubuntu release.
It only seems to occur with 720p when using fbdev. I will be reporting the bug shortly and once fixed Lakka can pull it in.

**Comment by hooby3dfx** [comment]
Fixed upstream!
https://github.com/hardkernel/linux/commit/bc94be360ae2652484afbd2c4a61b55b0a6d53ad

**Comment by Enverex** [comment]
I don't think that would fix it for me as I'm using 1280x800 and they seem to be hard-coding specific resolutions in... which seems an odd way of doing it.

**Comment by hooby3dfx** [comment]
Can you chime in on the odroid forum?

**Comment by hooby3dfx** [comment]
well at least this fixes 720p, which i think a lot of users have basically been blocked by.
@Kivutar can this go on your radar to pull in?

**Comment by Kivutar** [comment]
Yes, i'm updating it.
